### ISSUE TITLE
doesnt work on wayland

### ISSUE THREAD

**Issue by xyangst** [dimusic/active-win-pos-rs#28]
this crate does not work on wayland..
![image](https://github.com/dimusic/active-win-pos-rs/assets/93786592/95ccb0a7-85f3-4f3a-95b6-de59821b08d5)

**Comment by fdidron** [comment]
Do you have xwayland installed ? This library interfaces with xcb, Xwayland should give you a compatibility layer there.

**Comment by TukanDev** [comment]
> Do you have xwayland installed ? This library interfaces with xcb, Xwayland should give you a compatibility layer there.

I have same problem here i have xwayland installed and i get same error when i try to locate my dolphin window in focus or anything that is not using xwayland itself (example it works on discord, jetbrains apps...) but anything outside, it cant find. I do not know what is happening except that i have xwayland and it does not detect anything that is not running under xwayland at all as already mentioned unless both OP and I are missing some dependacy on our system besides normal xwayland package...

**Comment by ShayBox** [comment]
Yeah, looks like this can't detect wayland programs, only xwayland ones. Maybe this is possible with a portal?

**Comment by TukanDev** [comment]
> Yeah, looks like this can't detect wayland programs, only xwayland ones. Maybe this is possible with a portal?
> 
> There's an extension which could be adopted, but not a lot of compositors support it. https://github.com/activitywatch/aw-watcher-window-wayland

Im suprised wayland itself still does not seem to have a protocol for this.... even if it would be under ext category of them (atleast as far as i know it does not have any standard protocol) while compositors themself might have one for their use... Would be annoying to implement for every single one but most common ones should not be an issue.

However github you linked works on like 4 of them and we dont talk about GNOME thay are special... but if it doesnt work for wlroots based compositors and stuff like upocoming cosmic and unconfirmed kwin i would much more say lets just do it raw.

**Comment by ShayBox** [comment]
It has a protocol, but it's not adopted. Unfortunately this is how Wayland works, each compositor has to adopt the extension protocols. Kwin does not support it either, I tried.

**Comment by TukanDev** [comment]
> It has a protocol, but it's not adopted. Unfortunately this is how Wayland works, each compositor has to adopt the extension protocols. Kwin does not support it either, I tried. Afaik there's no way to do it without these protocols being supported, this is the security model of Wayland.

Yea sadly guess it just wont work on wayland for now...  which is sad to see as this is very useful library.

**Comment by ShayBox** [comment]
After looking into it again this seems to be the most active discussion following this use case, hopefully something can be done because the lack of active window information effectively kills my program for Wayland.

https://github.com/flatpak/xdg-desktop-portal/issues/304
Glad to see Cosmic becoming the forerunner in proposal adoption, hopefully something can come of this.

**Comment by TukanDev** [comment]
> After looking into it again this seems to be the most active discussion following this use case, hopefully something can be done because the lack of active window information effectively kills my program for Wayland.
> 
> [flatpak/xdg-desktop-portal#304](https://github.com/flatpak/xdg-desktop-portal/issues/304) Glad to see Cosmic becoming the forerunner in proposal adoption, hopefully something can come of this.
> 
> Sway, Wayfire, Cosmic, and maybe Hyprland(?) support it, I'm not sure about KDE, and Gnome is doing it's own thing as always.

Last time i tested this I was on KDE now I am on hyprland i should test this again and get back to you, these types of issues kinda made me stop working on my app, If we somehow get this working even on a fork of this for now it would be huge.
I love how flatpak issue you linked has like 5 years of bike shedding it seems about a very useful protocol.

**Comment by ImBIOS** [comment]
```
󰀵  󱑍 12:40  ﱮ active-win-pos-rs    main  is 📦 v0.9.1 via 🦀 v1.87.0   2s 
 ➜  cargo run --example active-window
   Compiling active-win-pos-rs v0.9.1 (/home/imbios/projects/active-win-pos-rs)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.18s
     Running `target/debug/examples/active-window`
active window: ActiveWindow {
    title: "cargo",
    process_path: "/opt/warpdotdev/warp-terminal/warp",
    app_name: "dev.warp.Warp",
    window_id: "50331653",
    process_id: 17532,
    position: WindowPosition {
        x: 2660.0,
        y: 489.0,
        width: 1280.0,
        height: 800.0,
    },
}
󰀵  󱑍 12:40  ﱮ active-win-pos-rs    main  is 📦 v0.9.1 via 🦀 v1.87.0 
 ➜  echo $XDG_SESSION_TYPE
wayland
```

it's working on wayland, I recommend closing this issue if no further activity as stale issue
